### PR TITLE
fix for issue 19309: does not pad tensor properly

### PR DESCRIPTION
**Pull request by azaytsev-epam**
### Ticket
[Link to Github Issue](https://github.com/tenstorrent/tt-metal/issues/19309)

### Problem description
Tensor layout conversion Tensor.to(Layout::TILE) throws instead of conversion
meanwhile ttnn.to_layout(tensor, layout=layout) works

### What's changed
- use the same logic for conversion
- Python regression test for the issue

### Checklist

- [ ] [All post-commit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml?query=branch:AlexeyZaytsev-epam/issue/19309/doesnot_pad_tensor_properly)

### CODE REVIEW

File: tests/ttnn/unit_tests/base_functionality/test_to_layout.py [hunk 1]
```diff
@@ -681,3 +681,27 @@ def test_shard_untilize2(device):
...
+    ],
+)
+@pytest.mark.parametrize("layout", [ttnn.ROW_MAJOR_LAYOUT, ttnn.TILE_LAYOUT])
+def test_tensor_padding(device, shape, layout):
+    if shape is None:
+        pt_tensor = torch.empty(0, dtype=torch.bfloat16, requires_grad=False)
+    else:
+        pt_tensor = torch.rand(torch.Size(shape), requires_grad=False).bfloat16()
+
+    tt_tensor_one = ttnn.to_layout(ttnn.Tensor(pt_tensor, ttnn.bfloat16), layout=layout).to(device)
```
Comment (aliaksei-sala) [inline tests/ttnn/unit_tests/base_functionality/test_to_layout.py]: I think the unit test should not depend on another operation (ttnn.to_layout).
Since the issue was related to padding, we can simply verify that .to(layout) correctly pads tensors by checking tensor shape

File: tests/ttnn/unit_tests/base_functionality/test_to_layout.py [hunk 2]
```diff
@@ -681,3 +681,24 @@ def test_shard_untilize2(device):
...
+)
+@pytest.mark.parametrize("layout", [ttnn.ROW_MAJOR_LAYOUT, ttnn.TILE_LAYOUT])
+def test_tensor_to(device, shape, layout):
+    if shape is None:
+        pt_tensor = torch.empty(0, dtype=torch.bfloat16, requires_grad=False)
+    else:
+        pt_tensor = torch.rand(torch.Size(shape), requires_grad=False).bfloat16()
+
+    output_tensor = ttnn.Tensor(pt_tensor, ttnn.bfloat16).to(layout)  # should not throw
+    assert output_tensor.layout == layout
```
Comment (aliaksei-sala) [inline tests/ttnn/unit_tests/base_functionality/test_to_layout.py]: Let's also verify the padded_shape() in the test

File: tests/ttnn/unit_tests/base_functionality/test_to_layout.py [hunk 3]
```diff
@@ -681,3 +682,38 @@ def test_shard_untilize2(device):
...
+
+    initial_shape = pt_tensor.shape  # store initial shape
+
+    output_tensor = ttnn.Tensor(pt_tensor, ttnn.bfloat16).to(layout)  # should not throw
+    result_shape = output_tensor.padded_shape  # store result shape
+
+    # Layout verification
+    assert output_tensor.layout == layout
+    # Padding verification: shape comparison
+    if ttnn.TILE_LAYOUT == layout:
```
Comment (aliaksei-sala) [inline tests/ttnn/unit_tests/base_functionality/test_to_layout.py]: Since the intent of this test is to validate padding behavior when calling .to(layout) for TILE layout, let’s give the test a more descriptive name.

test_tensor_to is too abstract and doesn’t clearly show what’s being tested. 

We can limit the test to the TILE layout only. Padding in the row-major layout also changes the logical shape, so it’s not padding in the general sense.

**Review by aliaksei-sala — APPROVED** [review]

File: tests/ttnn/unit_tests/base_functionality/test_to_layout.py (hunk 3 above)
Comment (azaytsev-epam) [reply tests/ttnn/unit_tests/base_functionality/test_to_layout.py]: @aliaksei-sala addressed

**Review by nsextonTT — APPROVED** [review]
LGTM

**Review by bbradelTT — APPROVED** [review]
